code/lib/cli/src/upgrade.ts

```diff
@@ -168,36 +181,79 @@ export const doUpgrade = async ({
-          packageJson.dependencies['react-scripts'] ?? packageJson.devDependencies['react-scripts'];
-        if (reactScriptsVersion && lt(coerceSemver(reactScriptsVersion), '5.0.0')) {
-          return false;
-        }
-      }
+  const { configDir: inferredConfigDir, mainConfig: mainConfigPath } = getStorybookInfo(
+    packageJson,
+    userSpecifiedConfigDir
+  );
+  const configDir = userSpecifiedConfigDir || inferredConfigDir || '.storybook';
+
+  let mainConfigLoadingError = '';
 
-      // only upgrade packages that are in the monorepo
-      return dependency in versions;
-    }) as Array<keyof typeof versions>;
-    return monorepoDependencies.map((dependency) => {
-      /* add ^ modifier to the version if this is the latest stable or prerelease version
-         example outputs: @storybook/react@^8.0.0 */
-      const maybeCaret = (!isOutdated || isPrerelease) && !isCanary ? '^' : '';
-      return `${dependency}@${maybeCaret}${versions[dependency]}`;
+  const mainConfig = await loadMainConfig({ configDir }).catch((err) => {
+    mainConfigLoadingError = String(err);
+    return false;
+  });
+
+  // GUARDS
+  if (!storybookVersion) {
+    logger.info(missingStorybookVersionMessage());
+    results = { preCheckFailure: PreCheckFailure.UNDETECTED_SB_VERSION };
+  } else if (
+    typeof mainConfigPath === 'undefined' ||
+    mainConfigLoadingError.includes('No configuration files have been found')
+  ) {
+    logger.info(mainjsNotFoundMessage(configDir));
+    results = { preCheckFailure: PreCheckFailure.MAINJS_NOT_FOUND };
+  } else if (typeof mainConfig === 'boolean') {
+    logger.info(mainjsExecutionFailureMessage(mainConfigPath, mainConfigLoadingError));
+    results = { preCheckFailure: PreCheckFailure.MAINJS_EVALUATION };
+  }
+
+  // BLOCKERS
+  if (!results && typeof mainConfig !== 'boolean' && typeof mainConfigPath !== 'undefined') {
+    const blockResult = await autoblock({
+      packageManager,
+      configDir,
+      packageJson,
+      mainConfig,
+      mainConfigPath,
     });
-  };
+    if (blockResult) {
+      results = { preCheckFailure: blockResult };
+    }
+  }
 
-  const upgradedDependencies = toUpgradedDependencies(packageJson.dependencies);
-  const upgradedDevDependencies = toUpgradedDependencies(packageJson.devDependencies);
+  // INSTALL UPDATED DEPENDENCIES
+  if (!dryRun && !results) {
+    const toUpgradedDependencies = (deps: Record<string, any>) => {
+      const monorepoDependencies = Object.keys(deps || {}).filter((dependency) => {
+        // don't upgrade @storybook/preset-create-react-app if react-scripts is < v5
+        if (dependency === '@storybook/preset-create-react-app') {
+          const reactScriptsVersion =
+            packageJson.dependencies['react-scripts'] ??
+            packageJson.devDependencies['react-scripts'];
+          if (reactScriptsVersion && lt(coerceSemver(reactScriptsVersion), '5.0.0')) {
+            return false;
+          }
+        }
```

Comment:
I think this isn't necessary anymore since it is handled in the `block-react-script-version` autoblock.